src/application/middlewares/authenticateToken.ts

```diff
@@ -13,8 +13,11 @@ export async function authenticateToken(
   res: Response,
   next: NextFunction,
 ): Promise<Response | void> {
-  const token = req.header('Authorization')?.replace('Bearer ', '');
+  let token = req.header('Authorization')?.replace('Bearer ', '');
```

Comment:
Consider consolidating the token retrieval logic to handle both headers and cookies in a single function for better maintainability.

Reply:
About this - currently its retrieving token from header, if no header provided it looks in cookies. Please clarify what should be changed in that logic

Reply:
Sorry, I mean, you can abstract the token retrieval into a separate function. This can help centralize the token retrieval logic and make it easier to manage and understand.

I would do suggest something like below:
```typescript
// some imports...

// Function to retrieve the token from the request
function getTokenFromRequest(req: Request): string | null {
  // Try to get token from Authorization header
  const authHeader = req.header('Authorization');
  if (authHeader && authHeader.startsWith('Bearer ')) {
    return authHeader.slice(7); // Remove 'Bearer ' from the start
  }

  // If no Authorization header, try getting the token from cookies
  return req.cookies['access_token'] || null;
}

export async function authenticateToken(
  req: RequestWithUser,
  res: Response,
  next: NextFunction,
): Promise<Response | void> {
  const token = getTokenFromRequest(req);

  if (!token) {
    return res
      .status(HttpStatusCode.UNAUTHORIZED)
      .json({ error: 'Unauthorized - Token missing' });
  }

  try {
    const decoded = jwt.verify(token, secretKey) as IUser;
    req.user = decoded;
    next();
  } catch (error) {
    return res
      .status(HttpStatusCode.UNAUTHORIZED)
      .json({ error: 'Unauthorized - Invalid token' });
  }
}

```

Reply:
And, maybe this change is far optional, but I think that this change will be crucial in respect to the quality reason.